python/thunder/rdds/series.py

```diff
@@ -561,6 +561,67 @@ def query(self, inds, var='inds', order='F', isOneBased=True):
 
         return keys, values
 
+    def meanOfRegion(self, keys):
+        """Takes the mean of Series values within a single region specified by the passed keys.
+
+        Parameters
+        ----------
+        keys: sequence of Series record keys
+
+        Returns
+        -------
+        tuple of ((mean of keys), (mean value))
+        """
+        bcRegionKeys = self.rdd.context.broadcast(frozenset(keys))
+        n, kmean, vmean = self.rdd.filter(lambda (k, v): k in bcRegionKeys.value) \
+            .map(lambda (k, v):  (array(k, dtype=v.dtype), v)) \
+            .aggregate(_MeanCombiner.createZeroTuple(),
+                       _MeanCombiner.mergeIntoMeanTuple,
+                       _MeanCombiner.combineMeanTuples)
+        kmean = tuple(kmean.astype('int32'))
+        return kmean, vmean
+
+    def meanByRegions(self, nestedKeys):
```

Comment:
Let's change to `meanByRegion` by analogy to `reduceByKey` (as opposed to `reduceByKeys`)